src/index.ts

```diff
@@ -224,11 +224,31 @@ export default class Mnoga {
    */
   public setPhrases(locale: string | string[], phrases: Phrases): void {
     const locales = getCanonicalLocales(locale);
-    const clonedPhrases = this.clonePhrases(phrases);
 
     locales.forEach((l) => {
-      this.phrases[l] = clonedPhrases;
+      this.phrases[l] = {};
     });
+    mergePhrases(locales.map(l => this.phrases[l]), phrases);
```

Comment:
If I put an invalid key in then I will get an error and will have wiped out all the locale store. I presume we would want a transaction to occur, which is really difficult to do with mutable data. 🤔 I presume we might want to perform a dry run and then an actual run to make sure there are no errors.

Reply:
So we'd first validate the keys and then do the actual merging? I think that could work. Alternatively, do we feel there is still a need to validate the keys at runtime? We could have it be part of linting / locale checks.

Reply:
> So we'd first validate the keys and then do the actual merging? I think that could work. Alternatively, do we feel there is still a need to validate the keys at runtime? We could have it be part of linting / locale checks.

If we're making it private I think we can be left to our devices.

But, I think having some runtime validations is nice because it does cover other edge cases. I think legacy uses mnoga for some things. Either way, it's not the end of the world if we want to just have a linting rule on the JSON files or simply allow anything that can be encoded as JSON to be used 🤷 The part I'm most concerned about is the delimiter, though. Allowing periods in keys would make it impossible to discover the phrase later on.

Reply:
Well, unless of course we used a flat JSON structure.